lte/gateway/c/session_manager/PipelinedClient.cpp

```diff
@@ -528,4 +527,20 @@ uint32_t AsyncPipelinedClient::get_current_teid() {
   return teid;
 }
 
+bool AsyncPipelinedClient::set_upf_node(
+    const std::string& node_id, const std::string& addr) {
+  upf_node_id      = node_id;
+  upf_node_ip_addr = addr;
+  MLOG(MINFO) << "Set_upf_node_id: " << upf_node_id;
+  MLOG(MINFO) << "Set_upf_n3_addr: " << upf_node_ip_addr;
+  return true;
+}
+
+std::string AsyncPipelinedClient::get_upf_n3_addr() {
+  return upf_node_ip_addr;
+}
+
+std::string AsyncPipelinedClient::get_upf_node_id() {
+  return upf_node_id;
+}
```

Comment:
These do not make sense to me at all. Why would sessiond set node id and ip address for UPF? Why is it trying to retrieve the same info then? Isn't UPF sending these info as part of node information? cc @panyogesh?

Reply:
You are right,  These information UPF is sending.  "set_upf_node" is called when we receive these info from UPF as part of initial Node message.  Now the qn -- where we do store this info ?  There is no other place, other than "Pipleline client,  for now 4g/5g client is same "PipelinedClient", In future we shud separate or inherit to create 4g pipelined object and 5g pipelined object separately. Even separating that we have issue while sending , so for now we are using one common Pipeliend client object.

Reply:
Venu, you should store this information as part of a session context or upf node context (if you want to add one). Not in an API object.

Reply:
I think you are  confused with node info vs session info ?  We already storing UPF local teid, keeping aside UPF N3 address, though we update N3 address to AMF/GnodeB.  I thought N3 address adding as part of session, we can do  once we start supporting multiple UPF nodes under single SMF session.

Reply:
I don't think there is any confusion. N3 context should be stored as part of the session context not as part of an API object.

Reply:
Now we moved this under SessionStateEnforcer.cpp